code/controllers/subsystems/holomap.dm

```diff
@@ -0,0 +1,113 @@
+// Minimap generation system adapted from vorestation, adapted from /vg/.
+// Seems to be much simpler/saner than /vg/'s implementation.
+
+//The structure holding a given map
+/datum/holomapdata
+	var/icon/holomap_base
+	var/list/icon/holomap_areas = list()
+	var/icon/holomap_combined
+	var/icon/holomap_areas_combined
+	var/icon/holomap_small
+
+SUBSYSTEM_DEF(minimap)
+	name = "Holomap"
+	flags = SS_NO_FIRE
+	init_order = SS_INIT_HOLOMAP
+
+	var/list/datum/holomapdata/holomaps = list()
+	var/list/station_holomaps = list()
+
+/datum/controller/subsystem/minimap/Initialize()
+	holomaps.len = world.maxz
+	for (var/z = 1 to world.maxz)
+		generateHolomap(z)
```

Comment:
How does this work with z-level creation, blank auxiliary z levels, shuttles?

Reply:
Since mid-round vg (lowercase) Z-Level creation were unusual back then, my original implementation doesn't really handle displaying holomaps of newly created Z-Levels. If you have a standardized proc called whenever a Z level is created mid-round, remember to have it call generateHolomap(z) after the generation is over.

This won't matter for Station Holomaps which only display the Station, Derelict, and Asteroid Z levels by default, but will for holomap uniform chips that will try and fetch the current Z level's image.